toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Utility to create and manipulate disks and partitions
+
+package configuration
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"microsoft.com/pkggen/internal/logger"
+)
+
+var (
```

Comment:
We should pass those variables as arguments or implement them as a class variables and the functions as the class methods. Depending on globals makes the code harder to test and reason about, more error prone and thread unsafe.